### PR TITLE
Add waiting room (player joined subscription) and dummy quiz creator

### PR DESCRIPTION
**Pull request by andmue**

### CODE REVIEW

File: frontend/src/components/pages/WebsocketUtils.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+export class WebsocketUtils {
+
+    public static websocketUrlByPath(path:string) {
+        return this.websocketProtocolByLocation() +
```
Comment (mleimer) [inline frontend/src/components/pages/WebsocketUtils.ts:4]: Evtl. Template-Syntax bevorzugen

File: frontend/src/components/pages/WebsocketUtils.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    private static websocketProtocolByLocation() {
+        return window.location.protocol === "https:" ? "wss://" : "ws://";
+    }
+
+    private static websocketPortWithColonByLocation() {
+        const defaultPort = window.location.protocol === "https:" ? "443" : "80";
+        // override for ionic dev server which does not proxy ws
+        if (window.location.port === "3000") {
+            return ":4000";
+        }
```
Comment (danielsuter) [inline frontend/src/components/pages/WebsocketUtils.ts]: arrrgh

File: frontend/src/components/pages/WebsocketUtils.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        else if (window.location.port !== defaultPort) {
```
Comment (mleimer) [inline frontend/src/components/pages/WebsocketUtils.ts]: Welcome to .NET?

File: frontend/src/components/pages/start/Create.tsx [hunk 4]
```diff
@@ -1,11 +1,102 @@
...
+        joinAsOperator(operatorId: $operatorId) {
+          operatorId
+          joinId
+        }
+    }
+`;
+
+class Create extends Component<any, any> {
+    state = {
+        quizState: "Quiz not created yet",
```
Comment (mleimer) [inline frontend/src/components/pages/start/Create.tsx]: single quotes?

File: frontend/src/index.tsx [hunk 5]
```diff
@@ -6,18 +6,37 @@ import * as serviceWorker from './serviceWorker';
...
+        const { kind, operation } = getMainDefinition(query);
+        return kind === 'OperationDefinition' && operation === 'subscription'
+    },
+    webSocketLink,
+    httpLink
+);
+
 const client: ApolloClient<NormalizedCacheObject> = new ApolloClient({
-    link: httpLink,
+    link: link,
```
Comment (danielsuter) [inline frontend/src/index.tsx]: write only once

**Review by danielsuter — APPROVED** [review]

**Pull request merged**